doc/site/docs/01-getting-started/03-shelf-vs-relic.md

```diff
@@ -0,0 +1,376 @@
+// Content-Type and encoding are part of the Body
+final response = Response.ok(
+  body: Body.fromString('<html>...</html>', mimeType: MimeType.html),
+);
+```
+
+This unified approach in Relic ensures the body content, encoding, content-type header, and content-length header stay in sync, reducing errors.
+
+## Routing
+
+### Basic Routing
+
+**Shelf** requires the separate `shelf_router` package:
+
+```dart
+import 'package:shelf/shelf.dart';
+import 'package:shelf_router/shelf_router.dart';
+
+final router = Router()
+  ..get('/users/<id>', (Request request, String id) {
+    return Response.ok('User $id');
+  });
+```
+
+**Relic** has routing built-in:
+
+```dart
+import 'package:relic/relic.dart';
+
+final router = Router<Handler>()
+  ..get('/users/:id', (RequestContext ctx) {
+    final id = ctx.pathParameters[#id];  // Symbol-based access
+    return ctx.respond(Response.ok(body: Body.fromString('User $id')));
+  });
+```
+
+### Routing Performance
+
+Shelf's router (from `shelf_router`) iterates through routes linearly—O(routes) complexity. Relic uses a [trie data structure](https://en.wikipedia.org/wiki/Trie) for O(segments) matching, which should be faster for applications with many routes.
+
+### Middleware on Routes
+
+**Shelf** applies middleware globally via `Pipeline`:
+
+```dart
+final handler = Pipeline()
+  .addMiddleware(logRequests())
+  .addMiddleware(authentication())
+  .addHandler(router);
+```
+
+**Relic** allows middleware at the route level with `router.use()`:
+
+```dart
+final router = Router<Handler>();
+
+// Apply middleware to specific paths
+router.use('/api', (handler) {
+  return (ctx) async {
+    // Auth check
+    if (!isAuthorized(ctx)) {
+      return ctx.respond(Response.forbidden());
+    }
+    return handler(ctx);
+  };
+});
+
+// Routes under /api automatically get the middleware
+router.get('/api/users', getUsersHandler);
+```
+
+## WebSocket Support
+
+### WebSocket Integration
+
+**Shelf** requires a separate package:
+
+```dart
+import 'package:shelf_web_socket/shelf_web_socket.dart';
+
+var handler = webSocketHandler((webSocket) {
+  webSocket.stream.listen((message) {
+    print('Received: $message');
+  });
+  webSocket.sink.add('Hello!');
+});
+```
+
+**Relic** has WebSockets built-in with state machine integration:
+
+```dart
+import 'package:relic/relic.dart';
+
+ConnectContext handler(NewContext ctx) {
+  return ctx.connect((RelicWebSocket ws) {
+    ws.events.listen((event) {
+      print('Received: $event');
+    });
+
+    // Non-throwing variant - returns false if connection closed
+    ws.trySendText('Hello!');
+
+    // Or use the throwing variant if you want exceptions
+    ws.sendText('Hello!');
+  });
+}
+```
+
+Relic's WebSocket API integrates with the context state machine and provides both throwing and non-throwing send methods for better error handling.
+
+## State Management
+
+### Request-Scoped State
+
+**Shelf** uses a context map with string keys:
```

Comment:
Perhaps also elaborate on the fact that you cannot have conflicts on keys with relic. Since `ContextProperty` objects are separate instances.

Reply:
Added: 

Since `ContextProperty` objects are separate instances, there's no risk of key conflicts between different properties, unlike string-based context maps.